### PR TITLE
Benchmark: Add  SxS V2/V3 capability

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Ability to run both V2 & V3 scenarios from same benchmark tool
- Support for stateful operations (easy to orchestrate). Once instance created per concurrency
- Both V2 & V3 targets
- BenchmarkNames through reflection
- Ability to code-inline for V3 SDK

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj [hunk 1]
```diff
@@ -16,7 +16,17 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="CommandLineParser" Version="2.5.0" />
-    <PackageReference Include="Microsoft.Azure.Cosmos" Version="*" />
     <PackageReference Include="HdrHistogram" Version="2.5.0" />
+    <PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="*" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(ProjectRef)' != 'True' ">
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj:23]: Enables easier master integration.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj [hunk 2]
```diff
@@ -16,7 +16,17 @@
...
+    <PackageReference Include="Microsoft.Azure.Cosmos" Version="*" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(ProjectRef)' == 'True' ">
+    <!--<PackageReference Include="Microsoft.Azure.Cosmos.Direct" Version="*" />
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj:28]: Uncomment to inline V3 code bits for latest changes

Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj:28]: Can we add these comments into the file?

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs [hunk 3]
```diff
@@ -87,6 +100,46 @@ internal static BenchmarkConfig From(string[] args)
...
         }
 
+        internal CosmosClient CreateCosmosClient(string accountKey)
+        {
+            CosmosClientOptions clientOptions = new CosmosClientOptions()
+            {
+                ApplicationName = BenchmarkConfig.UserAgentSuffix,
+                RequestTimeout = new TimeSpan(1, 0, 0),
+                MaxRetryAttemptsOnRateLimitedRequests = 0,
+                MaxRetryWaitTimeOnRateLimitedRequests = TimeSpan.FromSeconds(60),
```
[the remark this answers is not in the document]
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs:110]: Actually will exclude for consistency.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs [hunk 4]
```diff
@@ -7,133 +7,180 @@ namespace CosmosBenchmark
...
+                BenchmarkConfig config = BenchmarkConfig.From(args);
+                ThreadPool.SetMinThreads(config.MinThreadPoolSize, config.MinThreadPoolSize);
+                TelemetrySpan.IncludePercentile = config.EnableLatencyPercentiles;
+
+                string accountKey = config.Key;
+                config.Key = null; // Don't print
+                config.Print();
+
+                Program program = new Program();
+                await program.ExecuteAsync(config, accountKey);
```
Comment (j82w) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:42]: How do you exclude the prepare time from the operation? I don't see any iteration setup attribute.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:42]: its 10% exclusion always

**Pull request merged**